icelake/src/types/mod.rs

```diff
@@ -1,6 +1,8 @@
 //! Types will provide the definition of iceberg in-memory data types and
 //! functions to parse from on-disk files.
 
+#![allow(dead_code)]
```

Comment:
Why this?

Reply:
Seems there are deprecate fields but we don't use it now. Let's remove it.